File: Trie_of_blooming_directed_graphs_with_agents_and_filters/java/Root.java
package rootanchor;

import java.util.HashMap;
import java.util.Map;

/**
 * Root anchor: Trie_of_blooming_directed_graphs_with_agents_and_filters
 * Single entry point for blooming-directed-graph + filter_agents + trigger_process_agent per key.
 */
public final class Root {

    public static final class TrieNodeValue {
        public Object graphRef;
        public Object filterAgentsRef;
        public Object triggerProcessAgentRef;
    }

    public static final class TrieOfBloomingDirectedGraphsWithAgentsAndFilters {
        private final Map<Character, TrieOfBloomingDirectedGraphsWithAgentsAndFilters> children = new HashMap<>();
        private boolean isEnd;
        private TrieNodeValue value;

        public void insert(String key, TrieNodeValue val) {
            if (key == null || key.isEmpty()) {
                isEnd = true;
                value = val;
                return;
            }
            char c = key.charAt(0);
            children.computeIfAbsent(c, k -> new TrieOfBloomingDirectedGraphsWithAgentsAndFilters())
                    .insert(key.substring(1), val);
        }

        public TrieNodeValue get(String key) {
            if (key == null || key.isEmpty())
                return isEnd ? value : null;
            TrieOfBloomingDirectedGraphsWithAgentsAndFilters child = children.get(key.charAt(0));
            return child != null ? child.get(key.substring(1)) : null;
        }

        public boolean startsWith(String prefix) {
            if (prefix == null || prefix.isEmpty()) return true;
            TrieOfBloomingDirectedGraphsWithAgentsAndFilters child = children.get(prefix.charAt(0));
            return child != null && child.startsWith(prefix.substring(1));
        }
    }

    /** Global root anchor (Trie_of_blooming_directed_graphs_with_agents_and_filters). */
    public static final TrieOfBloomingDirectedGraphsWithAgentsAndFilters Trie_of_blooming_directed_graphs_with_agents_and_filters =
            new TrieOfBloomingDirectedGraphsWithAgentsAndFilters();

    private Root() {}
}

File: Trie_of_blooming_directed_graphs_with_agents_and_filters/swift/Root.swift
// Root anchor: Trie_of_blooming_directed_graphs_with_agents_and_filters
// Single entry point for blooming-directed-graph + filter_agents + trigger_process_agent per key.

import Foundation

/// Trie node holding a blooming-directed-graph reference plus agents and filters.
public struct TrieNodeValue {
    public var graphRef: Any?
    public var filterAgentsRef: Any?
    public var triggerProcessAgentRef: Any?
    public init(graphRef: Any? = nil, filterAgentsRef: Any? = nil, triggerProcessAgentRef: Any? = nil) {
        self.graphRef = graphRef
        self.filterAgentsRef = filterAgentsRef
        self.triggerProcessAgentRef = triggerProcessAgentRef
    }
}

/// Root anchor: Trie of blooming-directed-graphs with agents and filters (key = repo/context prefix).
public final class TrieOfBloomingDirectedGraphsWithAgentsAndFilters {
    private var children: [Character: TrieOfBloomingDirectedGraphsWithAgentsAndFilters] = [:]
    private var isEnd: Bool = false
    private var value: TrieNodeValue?

    public init() {}

    public func insert(key: String, value: TrieNodeValue) {
        var node: TrieOfBloomingDirectedGraphsWithAgentsAndFilters = self
        for char in key {
            if node.children[char] == nil {
                node.children[char] = TrieOfBloomingDirectedGraphsWithAgentsAndFilters()
            }
            node = node.children[char]!
        }
        node.isEnd = true
        node.value = value
    }

    public func get(key: String) -> TrieNodeValue? {
        var node: TrieOfBloomingDirectedGraphsWithAgentsAndFilters? = self
        for char in key {
            guard let next = node?.children[char] else { return nil }
            node = next
        }
        return node?.isEnd == true ? node?.value : nil
    }

    public func starts(with prefix: String) -> Bool {
        var node: TrieOfBloomingDirectedGraphsWithAgentsAndFilters? = self
        for char in prefix {
            guard let next = node?.children[char] else { return false }
            node = next
        }
        return true
    }
}

/// Global root anchor object (Trie_of_blooming_directed_graphs_with_agents_and_filters).
public let Trie_of_blooming_directed_graphs_with_agents_and_filters = TrieOfBloomingDirectedGraphsWithAgentsAndFilters()
